tests/test_modelwriter.py

```diff
@@ -144,3 +144,24 @@ def test_mw_error_2():
     writer = ModelWriter()
     with pytest.raises(ParameterWriterError):
         writer._write_changes("[not/a/path]")
+
+
+def test_write_mw_error_3(fileutils):
+    test_dir = fileutils.make_test_dir()
+
+    param_dict = {
+        "5": 10,  # MOM_input
+    }
+
+    conf_path = get_gen_file(fileutils, "easy/marked/")
+
+    # copy confs to gen directory
+    dir_util.copy_tree(conf_path, test_dir)
+    assert path.isdir(test_dir)
+
+    # init modelwriter
+    writer = ModelWriter()
+    with pytest.raises(SmartSimError):
```

Comment:
I know this wasn't a decision made as part of this PR, but is there a reason the `make_missing_tags_fatal=True` raises a generic `SmartSimError` over a `ParameterWriterError`?

Reply:
Nope, I think this was an old decision.

Reply:
I thought so. Don't change it with this PR, but I may check with the team to see if there is interest to using the more specific error in this/a future release.